src/activity_handlers.cpp

```diff
@@ -689,9 +700,19 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
             liquid.charges = item::INFINITE_CHARGES;
             break;
         case LST_MAP_ITEM:
-            if( static_cast<size_t>( act.values.at( 1 ) ) >= source_stack.size() ) {
+            if( static_cast<size_t>( act.values.at( 1 ) ) >= source_stack.size() ){
                 throw std::runtime_error( "could not find source item on ground for liquid transfer" );
             }
+            break;
+        case LST_MONSTER:
+        Creature *c = g->critter_at(source_pos);
+        source_mon = dynamic_cast<monster *>(c);
+        if( source_mon == nullptr ){
+            throw std::runtime_error( "could not find source creature for liquid transfer" );
+        }
+        liquid.deserialize( act.str_values.at( 0 ) );
+        break;
+
             on_ground = source_stack.begin();
```

Comment:
I thought I commented here but I don't know where that comment went, guess I don't know how github works yet.

Regarding the lines, I don't really know how the whole function works, including why it serialized/deserialized things, I have the vague idea that they have to do with checking the json for the liquid, but not much really.

Also already asked help in a review about a part of the code in that function. But then I don't know if that's how you ask for help or how you point out to what you have a problem with.